src/channel.js

```diff
@@ -1,46 +1,78 @@
 import {supabase} from './supabase-client.js'
+import {getUser} from './user.js'
 
-export const createChannel = async ({channel, user}) => {
-	const {name, slug} = channel
-	const {id: user_id} = user
+/**
+ * A channel
+ * @typedef {Object} Channel
+ * @property {string} name
+ * @property {string} slug - unique
+ * @property {string} [description]
+ */
+
+/**
+ * Creates a new radio channel and connects it to a user
+ * @param {Channel} fields
+ * @return {Promise<object>} {data, error}
+ */
+export const createChannel = async ({name, slug}) => {
+	const user = await getUser()
 
 	// Throw an error if the slug is in use by the old Firebase database.
 	const queryFirebaseDb = await firebaseGetChannelBySlug(slug)
 	const isSlugTaken = Object.keys(queryFirebaseDb).length > 0
-	if (isSlugTaken) throw new Error('Sorry. This channel slug is already taken by someone else.')
+	if (isSlugTaken) return {error: Error('Sorry. This channel slug is already taken by someone else.')}
 
 	// Create channel
-	const channelResponse = await supabase.from('channels').insert({name, slug}).single()
+	const channelRes = await supabase.from('channels').insert({name, slug}).single().select()
 
 	// Stop if the first query failed.
-	if (channelResponse.error) return channelResponse
+	if (channelRes.error) return channelRes
 
 	// Create junction table
-	const channel_id = channelResponse.data.id
-	const userChannelResponse = await supabase.from('user_channel').insert({user_id, channel_id}).single()
-	if (userChannelResponse.error) return userChannelResponse
+	const channel_id = channelRes.data.id
+	const userChannelRes = await supabase.from('user_channel').insert({user_id: user.id, channel_id}).single()
+	if (userChannelRes.error) return userChannelRes
```

Comment:
from supabase the `userChannelRes` has an `.error` and `.code` no?

Reply:
Yes. Here's an example

![image](https://user-images.githubusercontent.com/184567/196393370-292c6557-35f6-4123-b3e3-cbc0b931bf76.png)